### ISSUE TITLE
Grid Accessibility Issues

### ISSUE THREAD

**Issue by ericmahlberg** [IgniteUI/igniteui-angular#6341]
## Description  
The following accessibility bugs are present in the Grid's child controls.  I have been unable to override or find a workaround to them.

 * igniteui-angular version: 8.2.11
 * browser: Chrome

1) igx-grid-toolbar

Title: WCAG 1.3.1: Certain ARIA roles must contain particular children (igx-grid-toolbar)
Tags: Accessibility, WCAG 1.3.1, aria-required-children
Issue: Certain ARIA roles must contain particular children (aria-required-children - https://dequeuniversity.com/rules/axe/3.3/aria-required-children?application=msftAI)

Element path: igx-grid-toolbar
Snippet: <igx-grid-toolbar role="rowgroup" class="igx-grid-toolbar--compact ng-star-inserted" style="flex-basis: 851px;">

How to fix: 
Fix any of the following:
  Required ARIA child role not present: row

2) igx-grid__tfoot
Title: WCAG 1.3.1: Certain ARIA roles must contain particular children (.igx-grid__tfoot)
Tags: Accessibility, WCAG 1.3.1, aria-required-children
Issue: Certain ARIA roles must contain particular children (aria-required-children - https://dequeuniversity.com/rules/axe/3.3/aria-required-children?application=msftAI)

Element path: .igx-grid__tfoot
Snippet: <div class="igx-grid__tfoot" role="rowgroup" style="height: 0px;"><div class="igx-grid__tfoot-thumb" style="width: 17px; height: 0px;"></div></div>

How to fix: 
Fix any of the following:
  Required ARIA child role not present: row

3) classic grid filter
Title: WCAG 1.3.1,WCAG 3.3.2: Form elements must have labels (input[autocomplete="off"])
Tags: Accessibility, WCAG 1.3.1, WCAG 3.3.2, label
Issue: Form elements must have labels (label - https://dequeuniversity.com/rules/axe/3.3/label?application=msftAI)

Element path: input[autocomplete="off"]
Snippet: <input autocomplete="off" igxinput="" tabindex="0" placeholder="Add filter value" type="text" class="igx-input-group__input">

How to fix: 
Fix any of the following:
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Form element does not have an implicit (wrapped) <label>
  Form element does not have an explicit <label>
  Element has no title attribute or the title attribute is empty

4) igx-date-picker__input-date
Title: WCAG 1.3.1,WCAG 3.3.2: Form elements must have labels (.igx-date-picker__input-date)
Tags: Accessibility, WCAG 1.3.1, WCAG 3.3.2, label
Issue: Form elements must have labels (label - https://dequeuniversity.com/rules/axe/3.3/label?application=msftAI)

Element path: .igx-date-picker__input-date
Snippet: <input _ngcontent-wvh-c12="" class="igx-date-picker__input-date igx-input-group__input" igxinput="" type="text" placeholder="M/d/yyyy">

How to fix: 
Fix any of the following:
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Form element does not have an implicit (wrapped) <label>
  Form element does not have an explicit <label>
  Element has no title attribute or the title attribute is empty

5) igx-calendar-n
Title: WCAG 1.3.1: Certain ARIA roles must contain particular children (#igx-calendar-0)
Tags: Accessibility, WCAG 1.3.1, aria-required-children
Issue: Certain ARIA roles must contain particular children (aria-required-children - https://dequeuniversity.com/rules/axe/3.3/aria-required-children?application=msftAI)

Element path: #igx-calendar-n
Snippet: <igx-calendar _ngcontent-wvh-c16="" class="ng-tns-c17-12 igx-calendar" tabindex="0" id="igx-calendar-0" role="grid" aria-labelledby="calendar">

How to fix: 
Fix any of the following:
  Required ARIA children role not present: rowgroup row

Title: WCAG 4.1.1,WCAG 4.1.2: ARIA attributes must conform to valid values (#igx-calendar-0)
Tags: Accessibility, WCAG 4.1.1, WCAG 4.1.2, aria-valid-attr-value
Issue: ARIA attributes must conform to valid values (aria-valid-attr-value - https://dequeuniversity.com/rules/axe/3.3/aria-valid-attr-value?application=msftAI)

Element path: #igx-calendar-n
Snippet: <igx-calendar _ngcontent-wvh-c16="" class="ng-tns-c17-12 igx-calendar" tabindex="0" id="igx-calendar-0" role="grid" aria-labelledby="calendar">

How to fix: 
Fix all of the following:
  Invalid ARIA attribute value: aria-labelledby="calendar"

Title: WCAG 1.4.3: Elements must have sufficient color contrast (.igx-calendar__label:nth-child(1))
Tags: Accessibility, WCAG 1.4.3, color-contrast
Issue: Elements must have sufficient color contrast (color-contrast - https://dequeuniversity.com/rules/axe/3.3/color-contrast?application=msftAI)

Element path: .igx-calendar__label:nth-child(1)
Snippet: <span class="igx-calendar__label ng-star-inserted"> Sun </span>

How to fix: 
Fix any of the following:
  Element has insufficient color contrast of 1.88 (foreground color: #bdbdbd, background color: #ffffff, font size: 9.8pt (13px), font weight: normal). Expected contrast ratio of 4.5:1

6) #igx-checkbox-n-label
Title: WCAG 1.3.1,WCAG 4.1.1,WCAG 4.1.2: ARIA roles used must conform to valid values (#igx-checkbox-9-label)
Tags: Accessibility, WCAG 1.3.1, WCAG 4.1.1, WCAG 4.1.2, aria-roles

Issue: ARIA roles used must conform to valid values (aria-roles - https://dequeuniversity.com/rules/axe/3.3/aria-roles?application=msftAI)

Element path: #igx-checkbox-9-label
Snippet: <span role="label" class="igx-checkbox__label" id="igx-checkbox-9-label"></span>

How to fix: 
Fix all of the following:
  Role must be one of the valid ARIA roles

7) igx-checkbox-n-label
Title: WCAG 1.3.1,WCAG 4.1.1,WCAG 4.1.2: ARIA roles used must conform to valid values (#igx-checkbox-18-label)
Tags: Accessibility, WCAG 1.3.1, WCAG 4.1.1, WCAG 4.1.2, aria-roles

Issue: ARIA roles used must conform to valid values (aria-roles - https://dequeuniversity.com/rules/axe/3.3/aria-roles?application=msftAI)

Element path: #igx-checkbox-18-label
Snippet: <span role="label" class="igx-checkbox__label" id="igx-checkbox-18-label"> Core Columns Only </span>

How to fix: 
Fix all of the following:
  Role must be one of the valid ARIA roles
